### PR TITLE
LibDSP: Equalizer idk

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
I'll put a link to a good article on equalizer DSP theory when I find it again, for now rest assured that simple LSI systems like the one implemented in the first commit can easily support second-order band filters so a parametric multiband EQ is the current plan:tm:.

Also tests and a Lagom build of LibDSP (no I/O in library = no problem).

### CODE REVIEW

File: Userland/Libraries/LibAudio/Sample.h [hunk 1]
```diff
@@ -77,6 +79,18 @@ struct Sample {
         return log(val / VOLUME_A) / VOLUME_B;
     }
 
+    // Decibels are a logarithmic "unit":
+    // 20 dB = 10x difference between two loudness levels.
+    static ALWAYS_INLINE float db_to_linear(float const val)
+    {
+        return pow(10.0f, val / 20.0f);
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibAudio/Sample.h:86]:
```suggestion
        return pow(10.0f, val * 0.05f);
```

File: Userland/Libraries/LibAudio/Sample.h [hunk 2]
```diff
@@ -124,6 +124,23 @@ struct Sample {
...
+    {
+        left /= mult.left;
+        right /= mult.right;
+        return *this;
+    }
+
+    constexpr Sample operator/(Sample const mult) const
+    {
+        return { left / mult.left, right / mult.right };
+    }
```
[the remark this answers is not in the document]
Comment (kleinesfilmroellchen) [reply Userland/Libraries/LibAudio/Sample.h:156]: copy-pasta mistake

File: Userland/Libraries/LibAudio/Sample.h (hunk 1 above)
Comment (kleinesfilmroellchen) [reply Userland/Libraries/LibAudio/Sample.h:86]: Is this more precise? The division by 20 is much more intuitive especially if you don't know how the Decibel scale works.

Comment (Hendiadyoin1) [reply Userland/Libraries/LibAudio/Sample.h:86]: Well it is a division in a possibly hot code path,
Accuracy should be the same for our ears

Also decibel is weird, in some cases, like with pure energy, it has multiplier ten, but in a lot of actual use cases it has a 20 multiplier, because somewhere down the line there's a sqrt in the formulas....

Tldr: I'd say your choice, mult is faster division is more clear, in doubt comments are always an option

Comment (kleinesfilmroellchen) [reply Userland/Libraries/LibAudio/Sample.h:86]: Actually, factor 10 is only used for dBV, i.e. voltage decibels. But we're not electrical engineers :^)
